CenterForOpenScience/osf.io
# Feature/Zotero Restrict Citations to Unfiled Items at Root Level[PLAT-730]

## Purpose
Zotero citations appear to be duplicated when a user connects their root library (All Documents ) to a project.  The desired OSF behavior is that references appear only once, and aren't displayed at both the root, and a collection, if applicable. 

## Changes

- Zotero has no API call to fetch unfiled items; calls to fetch the root return all items.  Zotero displays all citations at the root level, but because of our hierarchical display of zotero citations, this looks like a bug on our end. On the OSF-side, we now exclude items that have been filed in one or more collections, and only display unfiled items and collections at the root.  
- Second change involves only returning top-level items at both the root and within collections. To explain, you can have a citation, and can upload a pdf to that citation.  The citation is the top-level item and that PDF is a child.  Zotero is only returning top-level items in their UI, and this would bring our UI in line with that. 
- Pyzotero has been upgraded to the latest version to allow returning top-level items only within collections.

## Before:
Film.txt is an attachment stored on This is a film.
<img width="474" alt="screen shot 2018-03-16 at 12 52 22 pm" src="https://user-images.githubusercontent.com/9755598/37542567-97d7494e-292c-11e8-9d14-1a106149e1e0.png">

## After:
Unfiled items have been removed from the root, and we are no longer showing child items
<img width="462" alt="screen shot 2018-03-16 at 12 55 07 pm" src="https://user-images.githubusercontent.com/9755598/37542569-99d8db40-292c-11e8-8e83-202b05bed1d3.png">

For comparison, this is what Zotero's UI shows:
Root level (Zotero shows all top-level items, filed and unfiled at root).  This is a film is filed under Films collection, but root shows everything. The attachment on This is a film is not visible.
<img width="951" alt="screen shot 2018-03-16 at 12 52 36 pm" src="https://user-images.githubusercontent.com/9755598/37542620-c57564b2-292c-11e8-9f89-5962dae41f6a.png">

Within film collections folder:
<img width="980" alt="screen shot 2018-03-16 at 12 52 43 pm" src="https://user-images.githubusercontent.com/9755598/37542635-d58084cc-292c-11e8-8af9-ce7882d7c7c4.png">


## QA Notes

Follow Becca's instructions in the ticket, creating several nested folders with several citations apiece, and verify that only unfiled items show up at the root.  I would add some tests with the same citation appearing in multiple collections, and verify that the citation shows up in both collections, but is absent from the root.  Also add attachments to the citations in zotero and verify that they no longer appear in either root or collections.  Test across personal library and group libraries.

## Side Effects



## Ticket

https://openscience.atlassian.net/browse/PLAT-730
